### PR TITLE
Add EVP_PKEY_METHOD enumeration functions.

### PR DESCRIPTION
**Pull request by snhenson**
##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

This PR adds a way to enumerate public key methods. It adds a test to ensure the table is correctly ordered and adds a -public-key-methods switch to the list command.

### CODE REVIEW

File: apps/openssl.c [hunk 1]
```diff
@@ -540,6 +546,21 @@ static void list_pkey(void)
     }
 }
 
+static void list_pkey_meth(void)
+{
+    size_t i;
+
+    for (i = 0; i < EVP_PKEY_meth_get_count(); i++) {
```
Comment (richsalz) [inline apps/openssl.c]: maybe add a variable to hold that get-count result?

File: doc/man3/EVP_PKEY_meth_get_count.pod [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                              const EVP_PKEY_METHOD *meth);
+
+=head1 DESCRIPTION
+
+EVP_PKEY_meth_count() returns a count of the number of public key methods
+available: it includes standard methods and any methods added by the
+application.
+
+EVP_PKEY_meth_get0() returns the public key method B<idx>. The value of B<idx>
+runs from zero to EVP_PKEY_meth_get_count() - 1.
```
Comment (richsalz) [inline doc/man3/EVP_PKEY_meth_get_count.pod]: s/runs from/must be between zero and/ ?

File: doc/man3/EVP_PKEY_meth_get_count.pod [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+EVP_PKEY_meth_get_count, EVP_PKEY_meth_get0, EVP_PKEY_meth_get0_info - enumeratepublic key methods
+
+=head1 SYNOPSIS
+
+ #include <openssl/evp.h>
+
+ size_t EVP_PKEY_meth_get_count(void);
+ const EVP_PKEY_METHOD *EVP_PKEY_meth_get0(size_t idx);
+ void EVP_PKEY_meth_get0_info(int *ppkey_id, int *pflags,
```
Comment (richsalz) [inline doc/man3/EVP_PKEY_meth_get_count.pod]: if meth is the input parameter, shouldn't that be first?

File: apps/openssl.c (hunk 1 above)
Comment (snhenson) [inline apps/openssl.c]: OK

File: doc/man3/EVP_PKEY_meth_get_count.pod (hunk 3 above)
Comment (snhenson) [inline doc/man3/EVP_PKEY_meth_get_count.pod]: It should but the function already exists and this is just documenting it.

**Comment by snhenson** [comment]
Update: rebased, comments addressed and conflicts resolved.

**Pull request closed without merging**